From ab5b224d6c7d4d9f2a8169047f5d33f9e1076f73 Mon Sep 17 00:00:00 2001
From: Daniel Hillenbrand <codeworkx@cyanogenmod.org>
Date: Sat, 24 Nov 2012 16:05:47 -0500
Subject: [PATCH] implement noise suppression for phone calls (1/3)

Make the default value of in-call noise suppression configurable.

Fix the value type of def_noise_suppression

settings provider: add noise suppression

Conflicts:
	packages/SettingsProvider/res/values/defaults.xml
---
 core/java/android/provider/Settings.java                          | 8 ++++++++
 packages/SettingsProvider/res/values/defaults.xml                 | 3 +++
 .../src/com/android/providers/settings/DatabaseHelper.java        | 4 ++++
 3 files changed, 15 insertions(+)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 2aa946f..c299ff2 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -2793,6 +2793,13 @@ public static void setShowGTalkServiceStatusForUser(ContentResolver cr, boolean
         public static final String TTY_MODE = "tty_mode";
 
         /**
+         * Whether noise suppression is enabled. The value is
+         * boolean (1 or 0).
+         * @hide
+         */
+        public static final String NOISE_SUPPRESSION = "noise_suppression";
+
+        /**
          * Whether the sounds effects (key clicks, lid open ...) are enabled. The value is
          * boolean (1 or 0).
          */
@@ -3552,6 +3559,7 @@ public static void setShowGTalkServiceStatusForUser(ContentResolver cr, boolean
             DTMF_TONE_TYPE_WHEN_DIALING,
             HEARING_AID,
             TTY_MODE,
+            NOISE_SUPPRESSION,
             SOUND_EFFECTS_ENABLED,
             HAPTIC_FEEDBACK_ENABLED,
             POWER_SOUNDS_ENABLED,       // moved to global
diff --git a/packages/SettingsProvider/res/values/defaults.xml b/packages/SettingsProvider/res/values/defaults.xml
index 80c1412..0ee25bc 100644
--- a/packages/SettingsProvider/res/values/defaults.xml
+++ b/packages/SettingsProvider/res/values/defaults.xml
@@ -251,4 +251,7 @@
          1. Right
          2. Left -->
     <integer name="def_qs_quick_pulldown">0</integer>
+
+    <!-- Default value of Settings.System.NOISE_SUPPRESSION -->
+    <bool name="def_noise_suppression">false</bool>
 </resources>
diff --git a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
index fa43390..2574cb2 100644
--- a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
+++ b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
@@ -2576,6 +2576,10 @@ private void loadSystemSettings(SQLiteDatabase db) {
             // Set default tty mode
             loadSetting(stmt, Settings.System.TTY_MODE, 0);
 
+            // Set default noise suppression value
+            loadBooleanSetting(stmt, Settings.System.NOISE_SUPPRESSION,
+                    R.bool.def_noise_suppression);
+
             loadIntegerSetting(stmt, Settings.System.SCREEN_BRIGHTNESS,
                     R.integer.def_screen_brightness);
 
